CONTRIBUTING.md

```diff
@@ -1,141 +1,95 @@
-
-```
-git lfs install
-git lfs pull
-urbit -F zod -B "bin/solid.pill" -A "pkg/arvo"
-```
-
-## Git practice
+## Issues
 
-### Contributing
+The [GitHub tracker][issu] is our canonical source of truth around issues, bugs,
+performance problems and feature requests.  If you encounter any issues when
+developing on Urbit, feel free to submit a report about it here.
 
-The canonical source tree is the `master` branch of
-[https://github.com/urbit/urbit][repo].  You should typically branch off of
-`master` when commencing new work. Most pull requests should be merging into
-one of the `next/*` branches, depending on what part of the system the pull
-request is targeting.
+A good bug report, description of a crash, etc., should ideally be
+*reproducible*, with clear steps as to how another developer can replicate and
+examine your problem.  That said, this isn't always possible -- some bugs depend
+on having created a complicated or unusual state, or can otherwise simply be
+difficult to trigger again.
 
-Since we use GitHub, we request you contribute via a GitHub pull request. Tag
-the [maintainer][main] for the component. If you have a question for the
-maintainer, you can direct message them from your Urbit ship using that
-information.
+Your issue should thus at a minimum be *informative*.  The best advice here is
+probably "don't write bad issues," where "bad" is a matter of judgment and
+taste.  Issues that the maintainers don't judge to be sufficiently useful or
+informative may be closed.
 
-When contributing changes, via whatever means, make sure you describe them
-appropriately.  You should attach a reasonably high-level summary of what the
-changes are and what they do; reference any useful background material that may
-exist, e.g. a GitHub issue, a mailing list discussion, a UP, etc.  [Here][jbpr]
-is a good example of a pull request with a useful, concise description.
+Feature requests are welcome, but they should include sufficient detail and
+explanation, as well as a discussion of perceived benefits one could expect from
+them.  "It would be cool if.." probably does not, in itself, constitute a good
+feature request; instead, try to be specific about what you're requesting, and
+what your desired feature would accomplish.
 
-If your changes replace significant extant functionality, be sure to compare
-them with the thing you're replacing.  You may also want to cc reviewers,
-or other parties who might have a particular interest in what you're
-contributing.
+### Feature Branch Names
 
-[jbpr]: https://github.com/urbit/urbit/pull/1782
+Every branch that you intend to put up for review must adhere to the form
+`i/<N>/<...>`, where `<N>` is the number of the issue that the branch
+corresponds to and `<...>` is an optional short description of the branch to aid
+in readability. If `<...>` is omitted, the `/` should be omitted as well, which
+makes `i/<N>` a well-formed branch name. These feature branches should be based
+off of `develop`.
```

Comment:
As mentioned out of band, not a huge fan of this. I could be convinced otherwise, but presently unclear what this gains us over "post the branch name to the issue when work has started", while adding the cost of more difficult to remember/work with branch names. Do we really care that much about having branch->issue lookup?

Reply:
I think this is actually important. It does two things, it forces an issue to exist in order to adhere to the branch naming spec and it makes it obvious what the branch is for that issue (can tab complete). CC @mcevoypeter who came up with the idea.